Panic when running `ec2ctl status`

Observed on v0.1.2.

```console
$ ec2ctl status --regions ap-southeast-3
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x104808818]

goroutine 13 [running]:
github.com/frgrisk/ec2ctl/adapter/aws.GetDeployedInstances(0x14000034460, {0x1400048c850, 0xe}, 0x1400054ec60, {0x1048e38cf, 0x6}, {0x14000377dc0, 0x0, 0x2})
        /Users/hskiba/go/pkg/mod/github.com/frgrisk/ec2ctl@v0.1.2/adapter/aws/instance.go:155 +0xad8
created by github.com/frgrisk/ec2ctl/cmd.getAccountSummary in goroutine 1
        /Users/hskiba/go/pkg/mod/github.com/frgrisk/ec2ctl@v0.1.2/cmd/status.go:90 +0x224
```

--- a/adapter/aws/instance.go
+++ b/adapter/aws/instance.go
@@ -149,19 +149,27 @@ func GetDeployedInstances(c chan RegionSummary, region string, tags map[string]s
 
 	for _, res := range result.Reservations {
 		for _, inst := range res.Instances {
-			instance.ID = *inst.InstanceId
+			if inst.InstanceId != nil {
+				instance.ID = *inst.InstanceId
+			}
 			instance.Status = inst.State.Name
 			instance.Type = inst.InstanceType
-			instance.IP = *inst.PrivateIpAddress
-			instance.Hibernation = *inst.HibernationOptions.Configured
+			if inst.PrivateIpAddress != nil {
+				instance.IP = *inst.PrivateIpAddress
+			}
+			if inst.HibernationOptions != nil && inst.HibernationOptions.Configured != nil {
+				instance.Hibernation = *inst.HibernationOptions.Configured
+			}
 			instance.Region = region
-			instance.AZ = getInstanceAZ(resultStatus.InstanceStatuses, inst.InstanceId)
+			if inst.InstanceId != nil {
+				instance.AZ = getInstanceAZ(resultStatus.InstanceStatuses, inst.InstanceId)
+			}
 			instance.SpotInstanceType = ""
 			if inst.InstanceLifecycle == "" {
 				instance.Lifecycle = string(types.InstanceLifecycleOnDemand)
 			} else {
 				instance.Lifecycle = string(inst.InstanceLifecycle)
-				if inst.InstanceLifecycle == types.InstanceLifecycleTypeSpot {
+				if inst.InstanceLifecycle == types.InstanceLifecycleTypeSpot && inst.SpotInstanceRequestId != nil {
 					instance.SpotInstanceType = getSpotRequestType(spotDetail.SpotInstanceRequests, inst.SpotInstanceRequestId)
 				}
 			}

